File: detox/android/detox/src/main/java/com/wix/detox/DetoxIdlePolicyConfig.java
package com.wix.detox;

import java.util.concurrent.TimeUnit;

import androidx.test.espresso.IdlingPolicies;

/**
 * Specification of values to use for Espresso's {@link IdlingPolicies} timeouts.
 * <br/>Overrides Espresso's defaults as they tend to be too short (e.g. when running on heavy-load app
 * on suboptimal CI machines).
 */
public class DetoxIdlePolicyConfig {
    /** Directly binds to {@link IdlingPolicies#setMasterPolicyTimeout(long, TimeUnit)}. Applied in seconds. */
    public Integer masterTimeoutSec = 120;
    /** Directly binds to {@link IdlingPolicies#setIdlingResourceTimeout(long, TimeUnit)}. Applied in seconds. */
    public Integer idleResourceTimeoutSec = 60;

    void apply() {
        IdlingPolicies.setMasterPolicyTimeout(masterTimeoutSec, TimeUnit.SECONDS);
        IdlingPolicies.setIdlingResourceTimeout(idleResourceTimeoutSec, TimeUnit.SECONDS);
    }
}

File: detox/android/detox/src/main/java/com/wix/detox/config/DetoxIdlePolicyConfig.kt
package com.wix.detox.config

import androidx.test.espresso.IdlingPolicies
import java.util.concurrent.TimeUnit

/**
 * Specification of values to use for Espresso's {@link IdlingPolicies} timeouts.
 *
 * Overrides Espresso's defaults as they tend to be too short (e.g. when running a heavy-load app
 * on suboptimal CI machines).
 */
class DetoxIdlePolicyConfig {
    /** Directly binds to [IdlingPolicies.setMasterPolicyTimeout]. Applied in seconds.  */
    @JvmField var masterTimeoutSec = 240

    /** Directly binds to [IdlingPolicies.setIdlingResourceTimeout]. Applied in seconds.  */
    @JvmField var idleResourceTimeoutSec = 180

    fun apply() {
        IdlingPolicies.setMasterPolicyTimeout(masterTimeoutSec.toLong(), TimeUnit.SECONDS)
        IdlingPolicies.setIdlingResourceTimeout(idleResourceTimeoutSec.toLong(), TimeUnit.SECONDS)
    }
}
